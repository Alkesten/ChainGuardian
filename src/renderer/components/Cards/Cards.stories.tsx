import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text, boolean, number, select, object, array } from '@storybook/addon-knobs';
import { ValidatorCard, NodeCard } from "./Cards";

storiesOf('Validator', module).add('Validator Card', () => {
    
    const textArrayValue = array("text",["Return (ETH)","ROI"]);
    const numberValue = number("number value", 20);
    // const mainValue = text("value","+10%");
    return <div >
            <ValidatorCard value={numberValue} textArray={textArrayValue}/>
        </div>;
}).addDecorator(withKnobs);

storiesOf('Validator', module).add('Node Card', () => {
    
    const textArrayValue = array("text",["Beacon Node","www.beacon.ethereum.org"]);
    const numberValue = number("number value", 250);
    // const mainValue = text("value","+10%");
    return <div >
            <NodeCard onClick={()=>{console.log("")}} value={numberValue} textArray={textArrayValue}/>
        </div>;
}).addDecorator(withKnobs);