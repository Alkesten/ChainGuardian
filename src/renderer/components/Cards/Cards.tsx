import * as React from "react";
import BN from "bn.js";

export interface ICardProps {
    textArray: Array<string>;
    value: BN | number;
    
}
export interface INodeCardProps extends ICardProps{
    onClick: () => void;
}
export const ValidatorCard: React.FunctionComponent<ICardProps> = (
    props: ICardProps) => {

    if (props.textArray[1]==="ROI") {
        return(
            <div className="validator-card-container">
                <h5>{props.textArray[0]}</h5>
                <h1 className={props.value>=0 ? "plus":"minus"}>
                    {props.value>=0 ? "+":"-"}{props.value}%</h1>
                <h5 className={props.value>=0 ? "plus":"minus"}>{props.textArray[1]}</h5>
            </div>
        );
    } else {
        return(
            <div className="validator-card-container">
                <h5>{props.textArray[0]}</h5>
                <h1>{
                    props.textArray[1]==="ETH" ? 
                        props.value.toString().slice(1)
                        :
                        props.value
                }</h1>
                <h5>{props.textArray[1]}</h5>
            </div>
        );
    }
};
export const NodeCard: React.FunctionComponent<INodeCardProps> = (
    props: INodeCardProps) => {
    return(
        <div onClick={(): void=>{props.onClick();}} className="node-card-container">
            <h2>{props.textArray[0]}</h2>
            <span className="node-text">{props.textArray[1]}</span>
            <h2>{props.value}ms</h2>
            <span className="node-text time">Average Response Time</span>
        </div>
    );
};