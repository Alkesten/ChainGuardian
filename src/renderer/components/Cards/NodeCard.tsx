import * as React from "react";

export interface INodeCardProps {
    title: string;
    value: number | string;
    url: string;
    onClick: () => void;
}

export const NodeCard: React.FunctionComponent<INodeCardProps> = (
    props: INodeCardProps) => {
    return(
        <div onClick={(): void=>{props.onClick();}} className="node-card-container">
            <h2>{props.title}</h2>
            <span className="node-text">{props.url}</span>
            <h2>{props.value} ms</h2>
            <span className="node-text time">Average Response Time</span>
        </div>
    );
};