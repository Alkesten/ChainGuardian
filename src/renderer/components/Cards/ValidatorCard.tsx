import * as React from "react";
import BN from "bn.js";

export interface IValidatorCardProps {
    title: string;
    type: string;
    value: BN | number;
}

export const ValidatorCard: React.FunctionComponent<IValidatorCardProps> = (
    props: IValidatorCardProps) => {

    if (props.type==="ROI") {
        return(
            <div className="validator-card-container">
                <h5>{props.title}</h5>
                <h1 className={props.value>=0 ? "plus":"minus"}>
                    {props.value>=0 ? "+":"-"}{props.value}%</h1>
                <h5 className={props.value>=0 ? "plus":"minus"}>{props.type}</h5>
            </div>
        );
    } else {
        return(
            <div className="validator-card-container">
                <h5>{props.title}</h5>
                <h1>{
                    props.type==="ETH" ? 
                        props.value.toString().slice(1)
                        :
                        props.value
                }</h1>
                <h5>{props.type}</h5>
            </div>
        );
    }
};