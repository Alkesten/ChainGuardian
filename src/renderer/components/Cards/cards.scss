.stats-container{
    background: url(../assets/img/cards/Polygon.svg) no-repeat center center;
    height: 160px;
    width: 160px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    h5:nth-child(1){
        color: red;
    }
    h2:nth-child(2){
        color: green;
    }
    h5:nth-child(3){
        color: blue;
    }
}