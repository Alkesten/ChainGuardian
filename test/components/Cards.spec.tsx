import * as React from "react";
import * as renderer from "react-test-renderer";
import {ValidatorCard} from "../../src/renderer/components/Cards/ValidatorCard";
import {NodeCard} from "../../src/renderer/components/Cards/NodeCard";

describe("Cards", () => {
    it("renders correctly, ValidatorCard, ROI", () => {
        const tree = renderer
            .create(<ValidatorCard 
                value={20} 
                title="Return (ETH)"
                type="ROI"
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders correctly, ValidatorCard, ETH", () => {
        const tree = renderer
            .create(<ValidatorCard 
                value={0.1405} 
                title="Balance"
                type="ETH"
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders correctly, ValidatorCard, days", () => {
        const tree = renderer
            .create(<ValidatorCard 
                value={45.3} 
                title="Uptime"
                type="DAYS"
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders correctly, NodeCard", () => {
        const tree = renderer
            .create(<NodeCard 
                onClick={(): void=>{console.log("");}} 
                value={21} 
                title="BeaconNode"
                url="rocketsonic.hr"
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});