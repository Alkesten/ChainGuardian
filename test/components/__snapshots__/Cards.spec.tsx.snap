// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards renders correctly, NodeCard 1`] = `
<div
  className="node-card-container"
  onClick={[Function]}
>
  <h2>
    BeaconNode
  </h2>
  <span
    className="node-text"
  >
    rocketsonic.hr
  </span>
  <h2>
    21
     ms
  </h2>
  <span
    className="node-text time"
  >
    Average Response Time
  </span>
</div>
`;

exports[`Cards renders correctly, ValidatorStat, ETH 1`] = `
<div
  className="validator-card-container"
>
  <h5>
    Balance
  </h5>
  <h1>
    .1405
  </h1>
  <h5>
    ETH
  </h5>
</div>
`;

exports[`Cards renders correctly, ValidatorStat, ROI 1`] = `
<div
  className="validator-card-container"
>
  <h5>
    Return (ETH)
  </h5>
  <h1
    className="plus"
  >
    +
    20
    %
  </h1>
  <h5
    className="plus"
  >
    ROI
  </h5>
</div>
`;

exports[`Cards renders correctly, ValidatorStat, days 1`] = `
<div
  className="validator-card-container"
>
  <h5>
    Uptime
  </h5>
  <h1>
    45.3
  </h1>
  <h5>
    DAYS
  </h5>
</div>
`;
